In [141]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [142]:
data = pd.read_csv("/content/1776311302-P3-Car Market Trends Analysis with Car Dekho Data.csv")
data.head(6)

,Car_Name,Year,Selling_Price,Present_Price,Kms_Driven,Fuel_Type,Seller_Type,Transmission,Owner
0,ritz,2014,3.35,5.59,27000,Petrol,Dealer,Manual,0
1,sx4,2013,4.75,9.54,43000,Diesel,Dealer,Manual,0
2,ciaz,2017,7.25,9.85,6900,Petrol,Dealer,Manual,0
3,wagon r,2011,2.85,4.15,5200,Petrol,Dealer,Manual,0
4,swift,2014,4.60,6.87,42450,Diesel,Dealer,Manual,0
5,vitara brezza,2018,9.25,9.83,2071,Diesel,Dealer,Manual,0


In [143]:
data.tail(3)

,Car_Name,Year,Selling_Price,Present_Price,Kms_Driven,Fuel_Type,Seller_Type,Transmission,Owner
298,city,2009,3.35,11.0,87934,Petrol,Dealer,Manual,0
299,city,2017,11.50,12.5,9000,Diesel,Dealer,Manual,0
300,brio,2016,5.30,5.9,5464,Petrol,Dealer,Manual,0


In [144]:
data.describe()

,Year,Selling_Price,Present_Price,Kms_Driven,Owner
count,301.000000,301.000000,301.000000,301.000000,301.000000
mean,2013.627907,4.661296,7.628472,36947.205980,0.043189
std,2.891554,5.082812,8.644115,38886.883882,0.247915
min,2003.000000,0.100000,0.320000,500.000000,0.000000
25%,2012.000000,0.900000,1.200000,15000.000000,0.000000
50%,2014.000000,3.600000,6.400000,32000.000000,0.000000
75%,2016.000000,6.000000,9.900000,48767.000000,0.000000
max,2018.000000,35.000000,92.600000,500000.000000,3.000000


From the above analysis we can say that

  1Q From which manufacturing year to which manufacturing year vechicles are present in this data
  
  1ans the data here contains cars with manufacturing year from 2003 to 2018.

  2Q what is the lowest price to which a vechicle is sold
  
  2ans the lowest price to which a car is sold at INR 10,000/-

  3Q what is the highest price to which a vechicle is sold
  
  3ans the highest price to which a car is sold at INR 35,00,000/-

  4Q how many records are there in the data
  
  4ans there are 301 records in the data

In [145]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 301 entries, 0 to 300
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Car_Name       301 non-null    object 
 1   Year           301 non-null    int64  
 2   Selling_Price  301 non-null    float64
 3   Present_Price  301 non-null    float64
 4   Kms_Driven     301 non-null    int64  
 5   Fuel_Type      301 non-null    object 
 6   Seller_Type    301 non-null    object 
 7   Transmission   301 non-null    object 
 8   Owner          301 non-null    int64  
dtypes: float64(2), int64(3), object(4)
memory usage: 21.3+ KB


In [146]:
data.isnull().sum()

,0
Car_Name,0
Year,0
Selling_Price,0
Present_Price,0
Kms_Driven,0
Fuel_Type,0
Seller_Type,0
Transmission,0
Owner,0


5Q Are there any missing records in this data

5ans There are no missing records in the data. All 301 records have complete values.

In [147]:
data["Car_Name"].nunique()

98

6Q How many different vehicles are present in this data

6ans There are 98 different vehicles present in the data.

In [148]:
data["Car_Name"].value_counts().head(10)

,count
Car_Name,
city,26
corolla altis,16
verna,14
fortuner,11
brio,10
ciaz,9
i20,9
innova,9
grand i10,8


7Q Which is the most sold vehicle in this data

7ans The most sold vehicle in the data is the City, with 26 records.

In [149]:
(data["Fuel_Type"] == "CNG").sum()

np.int64(2)

8Q Does the database include any CNG vehicle? If yes how many of them are there

8ans Yes, the database includes CNG vehicles. There are 2 CNG vehicles in the data.

In [150]:
(data["Seller_Type"] == "Individual").sum()

np.int64(106)

9Q How many vehicles here are for sale from individuals directly

9ans There are 106 vehicles that are sold by individuals directly.

In [151]:
(data["Transmission"] == "Automatic").sum()

np.int64(40)

10Q Does this database contain auto transmission vehicles? If yes how many of them are there

10ans Yes, the database contains automatic transmission vehicles. There are 40 automatic vehicles.

In [152]:
(data["Owner"] == 1).sum()

np.int64(10)

11Q How many single person owned vehicles are there in this database

11ans There are 10 single owner vehicles in the database.

In [153]:
data['Depreciation'] = data['Present_Price'] - data['Selling_Price']

most_depreciated = data.loc[data['Depreciation'].idxmax()]
least_depreciated = data.loc[data['Depreciation'].idxmin()]

print("Most Depreciated:", most_depreciated['Car_Name'], "| Value:", round(most_depreciated['Depreciation'], 2), "Lakhs")
print("Least Depreciated:", least_depreciated['Car_Name'], "| Value:", round(least_depreciated['Depreciation'], 2), "Lakhs")

Most Depreciated: land cruiser | Value: 57.6 Lakhs
Least Depreciated: Honda Activa 4G | Value: 0.03 Lakhs


12Q Which is the most and least cost depreciated vehicle in data ?

12ans The most cost-depreciated vehicle is the "land cruiser" (value dropped by 57.6 Lakhs), and the least depreciated is the "Honda Activa 4G" (dropped by only 0.03 Lakhs).

In [154]:
data.groupby("Car_Name")["Depreciation"].mean().sort_values().head(10)

,Depreciation
Car_Name,
TVS Sport,0.040
Honda Activa 4G,0.045
Hero Passion X pro,0.050
Bajaj Avenger 150,0.050
Honda Dream Yuga,0.060
Yamaha FZ 16,0.070
Activa 4g,0.110
Hero Splender iSmart,0.115
Royal Enfield Bullet 350,0.120


13Q Which brands of vehicles are less affected by cost depreciation

13ans The vehicles with the lowest average depreciation are less affected by cost depreciation. The analysis shows that some two-wheelers have very low depreciation compared with cars.

In [155]:
data[["Year", "Present_Price", "Kms_Driven", "Selling_Price", "Owner", "Depreciation"]].corr()["Depreciation"].sort_values(ascending=False)

,Depreciation
Depreciation,1.000000
Present_Price,0.864902
Selling_Price,0.520881
Kms_Driven,0.333832
Owner,0.107415
Year,-0.333746


14Q Are there any factors which you feel affect the cost depreciation

14ans Yes, factors such as vehicle age, present price, selling price, kilometers driven, ownership and the vehicle model affect cost depreciation.

In [156]:
data["Age"] = data["Year"].max() - data["Year"]

data[["Selling_Price", "Age", "Kms_Driven"]].corr()["Selling_Price"]

,Selling_Price
Selling_Price,1.000000
Age,-0.236141
Kms_Driven,0.029187


15Q In general selling price is affected by age of vehicle and distance driven by vehicle, is it observable from data

15ans The data shows a weak negative relationship between vehicle age and selling price. However, kilometers driven has almost no correlation with selling price in this dataset, so its effect is not clearly observable.

In [157]:
print(data[data["Year"] > 2014][["Car_Name", "Year"]])

print("Number of vehicles manufactured after 2014:",
      (data["Year"] > 2014).sum())

          Car_Name  Year
2             ciaz  2017
5    vitara brezza  2018
6             ciaz  2015
7          s cross  2015
8             ciaz  2016
..             ...   ...
295           city  2015
296           city  2016
297           brio  2015
299           city  2017
300           brio  2016

[147 rows x 2 columns]
Number of vehicles manufactured after 2014: 147


16Q Can we get idea about newest vehicles i.e. after 2014 manufactured

16ans Yes, we can identify vehicles manufactured after 2014 by filtering the Year column. There are 147 such vehicles in the data.

In [158]:
two_wheelers = data.iloc[101:201]

print(two_wheelers)
print("Number of two wheelers:", len(two_wheelers))

                      Car_Name  Year  Selling_Price  Present_Price  \
101         UM Renegade Mojave  2017           1.70           1.82   
102                  KTM RC200  2017           1.65           1.78   
103          Bajaj Dominar 400  2017           1.45           1.60   
104  Royal Enfield Classic 350  2017           1.35           1.47   
105                  KTM RC390  2015           1.35           2.37   
..                         ...   ...            ...            ...   
196                  Activa 3g  2008           0.17           0.52   
197           Honda CB twister  2010           0.16           0.51   
198         Bajaj Discover 125  2011           0.15           0.57   
199             Honda CB Shine  2007           0.12           0.58   
200           Bajaj Pulsar 150  2006           0.10           0.75   

     Kms_Driven Fuel_Type Seller_Type Transmission  Owner  Depreciation  Age  
101        1400    Petrol  Individual       Manual      0          0.12    1  
1

17Q Can we find out data of only two wheelers from this data

17ans Yes, we can separate the two-wheeler data from the car data. There are 100 two-wheelers in this database.

In [159]:
two_wheelers[two_wheelers["Year"] == two_wheelers["Year"].min()][["Car_Name", "Year", "Selling_Price"]]

,Car_Name,Year,Selling_Price
189,Hero Super Splendor,2005,0.2


18Q Which is the oldest bike sold here

18ans The oldest bike sold here is the Hero Super Splendor, manufactured in 2005.

In [160]:
two_wheelers[two_wheelers["Year"] == two_wheelers["Year"].max()][["Car_Name", "Year", "Selling_Price"]]

,Car_Name,Year,Selling_Price
101,UM Renegade Mojave,2017,1.70
102,KTM RC200,2017,1.65
103,Bajaj Dominar 400,2017,1.45
104,Royal Enfield Classic 350,2017,1.35
109,Royal Enfield Classic 350,2017,1.20
126,Bajaj Avenger 220,2017,0.90
128,Honda CB Hornet 160R,2017,0.80
129,Yamaha FZ S V 2.0,2017,0.78
130,Honda CB Hornet 160R,2017,0.75
132,Bajaj Avenger 220,2017,0.75


19Q Which is the newest bike sold here

19ans The newest bikes sold here were manufactured in 2017. Several two-wheelers in the data belong to this manufacturing year.

In [161]:
two_wheelers["Car_Name"].value_counts().head(5)

,count
Car_Name,
Royal Enfield Classic 350,7
Royal Enfield Thunder 350,4
Bajaj Pulsar 150,4
Bajaj Avenger 220,3
Yamaha FZ S V 2.0,3


20Q Which is the most sold bike here

20ans The most sold bike in this data is the Royal Enfield Classic 350, with 7 records.

In [162]:
bike_unusual = two_wheelers[
    two_wheelers["Selling_Price"] > two_wheelers["Present_Price"]
]

print("Two wheelers sold above present price:")
print(bike_unusual[["Car_Name", "Present_Price", "Selling_Price"]])

print("Number of such two wheelers:", len(bike_unusual))

Two wheelers sold above present price:
Empty DataFrame
Columns: [Car_Name, Present_Price, Selling_Price]
Index: []
Number of such two wheelers: 0


21Q Do you find any deal in two wheelers which exceeded the general expectation? Can you find reason for it

21ans No, there are no two-wheelers in the data where the selling price is greater than the present price.

In [163]:
cars = data.drop(two_wheelers.index)

print("Car data:")
print(cars)

print("Number of cars:", len(cars))

Car data:
    Car_Name  Year  Selling_Price  Present_Price  Kms_Driven Fuel_Type  \
0       ritz  2014           3.35           5.59       27000    Petrol   
1        sx4  2013           4.75           9.54       43000    Diesel   
2       ciaz  2017           7.25           9.85        6900    Petrol   
3    wagon r  2011           2.85           4.15        5200    Petrol   
4      swift  2014           4.60           6.87       42450    Diesel   
..       ...   ...            ...            ...         ...       ...   
296     city  2016           9.50          11.60       33988    Diesel   
297     brio  2015           4.00           5.90       60000    Petrol   
298     city  2009           3.35          11.00       87934    Petrol   
299     city  2017          11.50          12.50        9000    Diesel   
300     brio  2016           5.30           5.90        5464    Petrol   

    Seller_Type Transmission  Owner  Depreciation  Age  
0        Dealer       Manual      0         

22Q Can we find out data of only cars from this data

22ans Yes, we can separate the car records from the two-wheeler records. There are 201 cars in the database.

In [164]:
cars[cars["Year"] == cars["Year"].min()][["Car_Name", "Year", "Selling_Price"]]

,Car_Name,Year,Selling_Price
37,800,2003,0.35
39,sx4,2003,2.25


23Q Which is the oldest car sold here

23ans The oldest cars sold here were manufactured in 2003. The vehicles include the 800 and SX4.

In [165]:
cars[cars["Year"] == cars["Year"].max()][["Car_Name", "Year", "Selling_Price"]]

,Car_Name,Year,Selling_Price
5,vitara brezza,2018,9.25


24Q Which is the newest car sold here

24ans The newest car sold here is the Vitara Brezza, manufactured in 2018.

In [166]:
car_unusual = cars[
    cars["Selling_Price"] > cars["Present_Price"]
]

print("Cars sold above present price:")
print(car_unusual[["Car_Name", "Present_Price", "Selling_Price"]])

print("Number of such cars:", len(car_unusual))

Cars sold above present price:
Empty DataFrame
Columns: [Car_Name, Present_Price, Selling_Price]
Index: []
Number of such cars: 0


25Q Do you find any deal in cars which exceeded the general expectation? Can you find reason for it

25ans No, there are no cars in the data where the selling price is greater than the present price. Therefore, no unusual deal of this type is observed.